mmengine/runner/runner.py

```diff
@@ -1600,6 +1605,19 @@ def train(self) -> nn.Module:
         Returns:
             nn.Module: The model after training.
         """
+        if is_model_wrapper(self.model):
+            ori_model = self.model.module
+        else:
+            ori_model = self.model
+        assert hasattr(ori_model, 'train_step'), (
+            'If you want to train your model, please make sure your model '
+            'has implemented `train_step`.')
+
+        if self._val_loop is not None:
+            assert hasattr(ori_model, 'val_step'), (
+                'If you want to validate your model, please make sure your '
+                'model has implemented `val_step`.')
+
```

Comment:
Suggest moving those checkings to `build_model` method.

Reply:
Since it is unknown that user will call `runner.train`,`runner.val` or `runner.test`,  `build_model` does not know to check `train_step`,`val_step` or `test_step`.

Reply:
got it